httpserver/http_connection.cpp

```diff
@@ -155,8 +158,26 @@ accept_connection(boost::asio::io_context& ioc,
                 ->start();
 
         if (ec) {
-            LOKI_LOG(error, "Could not accept a new connection {}: {}",
-                     ec.value(), ec.message());
+            LOKI_LOG(
+                error,
+                "Could not accept a new connection {}: {}. Will only start "
+                "accepting new connections after a short delay.",
+                ec.value(), ec.message());
+
+            // If we fail here we are unlikely to be able to accept a new
+            // connection immediately, hence the delay
+            acceptor_timer.expires_after(ACCEPT_DELAY);
+            acceptor_timer.async_wait([&](const error_code& ec) {
+                if (ec && ec != boost::asio::error::operation_aborted) {
+                    // Not sure how to recover here, so it is probably the
+                    // safest to simply abort and let the launcher/systemd
+                    // restart us
+                    abort();
+                }
+
+                accept_connection(ioc, ssl_ctx, acceptor, sn,
```

Comment:
Haven't been around the storage server code for a while but without removing the accept_connection call below (or putting it inside an else statement) won't this just immediately accept a connection and then accept *another* connection after the delay?

Reply:
I agree, I think that following `accept_connection` should be in an else.

Reply:
(Or it could be in the `!ec` case above it)

Reply:
Yep, that's a bug, fixed now.